### DISCUSSION TITLE
How to access L1 Batch block number?

### DISCUSSION THREAD

**Discussion by Sam-Scolari** [zkSync-Community-Hub/zksync-developers#188] [General] [answered]
With the completion of the upgrade plan for block.number and block.timestamp how do we now access the latest block of the L1 batch? The update post stated an alternative would be added but I have not been able to find it anywhere.

**Comment by MexicanAce** [comment, marked as answer]
Hi @Sam-Scolari 👋 !

This information is available via the `zks_L1BatchNumber ` API : https://era.zksync.io/docs/reference/concepts/blocks.html#retrieving-block-and-batch-numbers

**Comment by daisbryce** [comment]
To access the latest block of the L1 batch after the completion of the upgrade plan for block.number and block.timestamp, you can use the zksync.getBlockNumber method in the zkSync SDK. This method retrieves the latest block number of the zkSync chain, allowing you to access information about the most recent batch. Make sure to update your code to use this alternative method for retrieving block information.

**Comment by kaaron9** [comment]
The latest block of the L1 batch can now be accessed using blockHash. This provides an alternative method to retrieve information about the latest block after the upgrade plan for block.number and block.timestamp.

**Comment by poinsondrew** [comment]
Access the latest block of the L1 batch using the new alternative method.

**Comment by hollinsmic** [comment]
After the upgrade plan for block.number and block.timestamp, accessing the latest block of the L1 batch can be done using the new block.chainid variable. This variable returns the chain ID of the current chain and can be used to determine the latest block of the L1 batch.

**Comment by hilliardloyd** [comment]
To access the latest block of the L1 batch after the upgrade plan for block.number and block.timestamp, use the "eth_getBlockByNumber" JSON-RPC method with the parameter "latest".
